File: UniEatsBackEnd/Models/Payment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class Payment
    {
        [Key]
        public int PaymentId { get; set; }

        [Required]
        public int OrderId { get; set; }

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "Amount must be greater than 0.")]
        public decimal Amount { get; set; }

        [Required]
        [RegularExpression("(Card|Cash)", ErrorMessage = "Invalid payment method.")]
        public string PaymentMethod { get; set; }

        [Required]
        [RegularExpression("(Paid|Pending)", ErrorMessage = "Invalid payment status.")]
        public string PaymentStatus { get; set; } = "Pending";

        public string TransactionId { get; set; }

        [Required]
        public DateTime PaymentDate { get; set; } = DateTime.Now;
    }
}

File: UniEatsFrontEnd/UniEatsFrontEnd/src/models/payment.ts
import { Order } from './order';

export interface Payment {
  paymentId: number; // Unique identifier for the payment
  orderId: number; // ID of the associated order
  amount: number; // Payment amount, must be greater than 0
  paymentMethod: 'Card' | 'Cash' | 'Online' | 'GiftCard'; // Allowed payment methods
  paymentStatus: 'Paid' | 'Pending' | 'Refunded'; // Payment status, defaults to 'Pending'
  transactionId?: string; // Optional transaction ID, max length 100
  paymentDate?: Date; // Defaults to the current date
  refundAmount?: number; // Refund amount, defaults to 0
  order?: Order | null; // Optional navigation property for the associated order
}
